### PR TITLE
Ledge occupancy bugfix + rework

### LINKED ISSUE #232: ledge occupancy during ledgeroll is potentially nonfunctional
some reports out there, as well as my memory, that it looks like rolling from ledge doesnt hold ledge anymore. We'll need to look into ledge occupancy code for ledgeroll.

### LINKED ISSUE #279: Pyra and Charizard ledge trump
It is currently possible for Charizard and Pyra to be ledge trumped to during their ledge invincibility frames. [https://discordapp.com/channels/659964948365049887/660221102085111808/952817417757794355](url)
[https://discordapp.com/channels/659964948365049887/811729896161411092/952817346282651648](url)

### PR DESCRIPTION
**Pull request by jobrien97**
Ledge only counts as "occupied" for the first 3/4 of your ledge getup/roll/attack/jump animation (this will therefore vary slightly per character)

Fixes #232 
Fixes #279 
Related to #235

### CODE REVIEW

File: fighters/common/src/opff/ledges.rs [hunk 1]
```diff
@@ -136,27 +79,26 @@ unsafe fn occupy_ledge(boma: &mut BattleObjectModuleAccessor, status_kind: i32,
...
-            };
-
-            if i == player_number || VarModule::get_float(battle_object, vars::common::LEDGE_POS_X) == 0.0 {
-                continue;
-            }
-
-            if ledge_try_pos.x == VarModule::get_float(battle_object, vars::common::LEDGE_POS_X) && ledge_try_pos.y == VarModule::get_float(battle_object, vars::common::LEDGE_POS_Y) {
-                println!("Sending tethering player {} into ledge trump as they are trying to rewind to an occupied ledge.", player_number);
-                StatusModule::change_status_request_from_script(boma, *FIGHTER_STATUS_KIND_CLIFF_ROBBED, false);
+            if let Some(object_id) = ::utils::util::get_active_battle_object_id_from_entry_id(i) {
```
Comment (techyCoder81) [inline fighters/common/src/opff/ledges.rs:82]: has this been tested with ICs? In particular, this code as implemented might literally make nana ledgehog popo. Probably need to check if the entry id is the same as the entry id of the boma on ledge, or else if the boma on ledge matches the boma->partner field.

Comment (jobrien97) [reply fighters/common/src/opff/ledges.rs:82]: Confirmed Nana does not ledgehog Popo

**Review by techyCoder81 — DISMISSED** [review]
Just needs to be played in game a bunch at this point,  lgtm

**Comment by MadVillainLordQ** [comment]
Instead of full animation, it should be switched to 3/4 of the FAF. Currently ledge occupancy is at 43 frames with your implementation and a full ledge roll is 45 frames until actionability.

**Comment by jobrien97** [comment]
> Instead of full animation, it should be switched to 3/4 of the FAF. Currently ledge occupancy is at 43 frames with your implementation and a full ledge roll is 45 frames until actionability.

Just implemented this in latest commit

**Review by techyCoder81 — APPROVED** [review]

**Pull request merged**